### PR TITLE
OPENTOK-45029: added captions to meet

### PR DESCRIPTION
**Pull request by michaelwhittemore**
#### What is this PR doing?
Add publisher and session level UI. Update opentok angular to point to my branch with caption functionality 

#### How should this be manually tested?
Run meet locally. You'll need to re-run npm install because this makes some package changes. Then join a room in two different tabs (ensure they're both moderators). Speak and observe the captions being rendered. Change your name in the opentok text chat window and ensure that the name appears when you next speak in captions.

Captions backend is still flaky. This may take several tries. 

#### What are the relevant tickets?
OPENTOK-45030, [OPENTOK-45029](https://jira.vonage.com/browse/OPENTOK-45029)

### CODE REVIEW

File: plugins/transcribing/index.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+    const captionURL = `${apiUrl}/v2/project/${apiKey}/captions`;
+
+    const captionPostBody = {
+      sessionId: req.body.sessionId,
+      token: req.body.token,
+      languageCode: 'en-US',
+      maxDuration: 36000,
+      partialCaptions: 'true',
+      statusCallbackUrl: 'https://send-status-to.me',
```
Comment (v-kpheng) [inline plugins/transcribing/index.js:19]: Do we need `statusCallbackUrl`?

File: src/js/controllers.js [hunk 2]
```diff
@@ -161,6 +161,44 @@ angular.module('opentok-meet').controller('RoomCtrl', ['$scope', '$http', '$wind
       });
     };
 
+    const startTranscribing = () => {
+      $scope.transcribing = true;
+      const postBody = {
+        sessionId: $scope.session.sessionId,
+        token: $scope.session.token,
+      };
+      console.log(`postBody: ${postBody}`)
```
Comment (v-kpheng) [inline src/js/controllers.js:170]: Do we still need this console log?

File: src/js/directives.js [hunk 3]
```diff
@@ -1,4 +1,6 @@
 require('../css/cycle-camera.css');
+require('../css/toggle-captions.css');
+
```
Comment (v-kpheng) [inline src/js/directives.js:3]: Nit.  Superfluous newline.

**Review by v-kpheng — APPROVED** [review]
LGTM! 💪 🚀 

That said, left some nitpicks.

File: plugins/transcribing/index.js (hunk 1 above)
Comment (michaelwhittemore) [reply plugins/transcribing/index.js:19]: Good catch

File: src/js/controllers.js (hunk 2 above)
Comment (michaelwhittemore) [reply src/js/controllers.js:170]: I'll remove it

**Review by v-kpheng — APPROVED** [review]
LGTM! 💪 🚀 

That said, please add instructions on how to manually test.  Thanks! 🙏

File: plugins/transcribing/index.js [hunk 4]
```diff
@@ -52,13 +53,15 @@ module.exports = (app, config) => {
...
+      console.log(captionResponse)
     } catch (err) {
       console.warn(err);
       res.status(500);
       res.send(`Error stopping transcription services: ${err}`);
       return;
     }
 
-    res.sendStatus(captionResponse.status);
+    // Express send status is only supported on >=s 4.x
```
Comment (v-kpheng) [inline plugins/transcribing/index.js:61]: Nit: `>=s`.

**Review by v-kpheng — APPROVED** [review]
LGTM! 💪 🚀

File: plugins/transcribing/index.js (hunk 4 above)
Comment (michaelwhittemore) [reply plugins/transcribing/index.js:61]: Oops, looks like I didn't hit cmd-s when trying to save. Good catch.

**Review by v-kpheng — APPROVED** [review]
LGTM! 💪 🚀

**Comment by behei-vonage** [comment]
@michaelwhittemore, @cpettet, and I teamed up to get this tested as Chris and I are not able to run meet locally. We saw the captions come up only after the "captions" button was toggled a few times.

**Review by behei-vonage — APPROVED** [review]
Michael demonstrated that it works as expected locally on his computer as we were not able to get meet to run locally for myself and @cpettet. 🚀

**Pull request merged**